plasma_framework/docs/design/payment-game-implementation-v1.md

```diff
@@ -0,0 +1,111 @@
+# Payment ExitGame Implementation V1
+
+This document is a short description of the interaction functions of the current Payment ExitGame implementation. Please check the [ALD ExitGame definitions and properties doc](./ald-exit-game-definitions-and-properties.md) for the description of each specific phrases used in here.
+
+In Payment transaction V1, it only allows to have Payment transaction V1 as inputs and have both Payment transaction V1 and V2 as outputs. Specification of V2 is still undefined. However, the spending condition of V1 to V2 is defined as the signature of output owner should be correctly signed.
+
+## Transaction Data Structure
+```
+struct Transaction {
+    uint256 txType;
+    bytes32[] inputs;
+    Output[] outputs;
+    bytes32 metaData;
+}
+
+struct Output {
+    uint256 outputType;
+    bytes20 outputGuard;
+    address token;
+    uint256 amount;
+}
+```
+
+In the first version of our Plasma Framework system, there would be Payment Transaction running only. 
+1. `txType` represents the type of the transaction. Each transaction type would have one and only one exit game contract mapping the exit game interactions. 
+1. The `inputs` field would be an array of `bytes32` holding `utxoPos` as the output identifier. There would be at most 4 inputs. For deposit transaction it should be with no inputs.
+1. The `outputs` field would be an array of `Output` struct holding payment output information. There would be at most 4 outputs. For deposit transaction it should be with one and only one output.
+1. `metaData` field is a `bytes32` that can be used to put extra data for the transaction. It should be an empty bytes if the transaction does not want to hold any data in this field.
```

Comment:
```suggestion
1. `metaData` field is a `bytes32` field that can be used to add extra data to the transaction. It should contain zeros if the transaction does not have an extra data.
```

Reply:
> It should contain zeros if the transaction does not have an extra data.

awwwwww....one more potential encode/decode issue. We cannot differentiate bytes with zero and without from RLP.

Reply:
I think the encoded `metadata` binary field should be interpreted as `bytes32` after/during decoding, so with zero-padding. Not 100% sure though here.